assets/js/interactivity/directives.js

```diff
@@ -28,79 +42,96 @@ export default () => {
 	directive(
 		'context',
 		( {
-			directives: {
-				context: { default: context },
-			},
+			directives: { context },
 			props: { children },
-			context: inherited,
+			context: inheritedContext,
 		} ) => {
-			const { Provider } = inherited;
-			const inheritedValue = useContext( inherited );
-			const value = useMemo( () => {
-				const localValue = deepSignal( context );
-				mergeDeepSignals( localValue, inheritedValue );
-				return localValue;
-			}, [ context, inheritedValue ] );
-
-			return <Provider value={ value }>{ children }</Provider>;
+			return context.reduceRight(
+				( toReturn, { namespace, value: newContext } ) => {
+					const { Provider } = inheritedContext;
+					const inheritedValue = useContext( inheritedContext );
+					const currentValue = useRef( deepSignal( {} ) );
+					currentValue.current = useMemo( () => {
+						const newValue = deepSignal( {
+							[ namespace ]: newContext,
+						} );
+						mergeDeepSignals( newValue, inheritedValue );
+						mergeDeepSignals(
+							currentValue.current,
+							newValue,
+							true
+						);
+						return currentValue.current;
+					}, [ newContext, inheritedValue ] );
+
+					return (
+						<Provider value={ currentValue.current }>
+							{ toReturn }
+						</Provider>
```

Comment:
Why are you wrapping providers with the one another instead of reusing the same one?

Reply:
Completely right. I'll update that part.

Reply:
Fixed in https://github.com/woocommerce/woocommerce-blocks/pull/11071/commits/f0508c973da57f1d3f061e161865354b088c1aaa.